.github/workflows/ci.yaml

```diff
@@ -17,3 +17,83 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v3
+      - name: Unshallow
+        run: |
+          git fetch --prune --unshallow
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          config-inline: |
+            [worker.oci]
+              gc = true
+              gckeepstorage = 100000 # 100 GiB
+
+              [[worker.oci.gcpolicy]]
+                keepBytes = 32212254720 # 30 GiB
+                keepDuration = 604800
+                filters = [ "type==source.local", "type==exec.cachemount", "type==source.git.checkout"]
+              [[worker.oci.gcpolicy]]
+                all = true
+                keepBytes = 107374182400 # 100 GiB
+            [registry."registry.ci.svc.cluster.local:5000"]
+              http = true
```

Comment:
this will be a huge pain to work around in all projects, registries which have non-localhost hostnames are treated as secure by default, so I think it might be easier to make it secure registry now

Reply:
secure means we need to bring in certs information here too, not sure I follow the problem here